rust_src/src/fns.rs

```diff
@@ -306,4 +308,41 @@ pub extern "C" fn internal_equal_misc(
     }
 }
 
+/// Return list of 1 minute, 5 minute and 15 minute load averages.
+///
+/// Each of the three load averages is multiplied by 100, then converted
+/// to integer.
+///
+/// When USE-FLOATS is non-nil, floats will be used instead of integers.
+/// These floats are not multiplied by 100.
+///
+/// If the 5-minute or 15-minute load averages are not available, return a
+/// shortened list, containing only those averages which are available.
+///
+/// An error is thrown if the load average can't be obtained.  In some
+/// cases making it work would require Emacs being installed setuid or
+/// setgid so that it can read kernel information, and that usually isn't
+/// advisable.
+#[lisp_fn(min = "0")]
+pub fn load_average(use_floats: bool) -> LispObject {
```

Comment:
You can change the return type to `Vec<LispObject>` and drop the final `LispObject::from`. The `lisp_fn` macro uses magic to convert everything to `LispObject`. `memory_use_counts` in `alloc.rs` is another example of this.

Reply:
In this case, we know that the result will be a list of ints or a list of floats, so even `Vec<LispObject>` is too general. But I don't know how to make it more specific. Ideally converting a `Vec` to `LispObject` would recursively convert its items as well. Is there a way to do that? A generic type or something like that?

(That would be outside the scope of this PR of course.)

Reply:
I think we would need to write a bunch of `impl From<Vec<EmacsInt>> for LispObject` style code. Should not be too complicated. Might even be able to semi-automate it with a macro.

Reply:
The returned values are either floats or integers, so a type that describes that would probably have to be some wrapping type like `enum EmacsNumber { Double(EmacsDouble), Int(EmacsInt) }` which should then be able to again use `From` on the inner type.

Reply:
`LispNumber` is our wrapper for a thing that can be a float or int.

Reply:
Ah there it is. I have some changes locally that do let us return `Vec<LispNumber>` with what seems like a reasonable `impl From<Vec<LispNumber>> for LispObject` but that's getting out of scope for this change.

Reply:
yeah. If it is worth a follow on PR, then cool.